common/changes/@rushstack/heft-jest-plugin/user-danade-JestDisplayName_2021-09-01-00-38.json

```diff
@@ -0,0 +1,11 @@
+{
```

Comment:
> Additionally, the test duration value is not always available/provided by Jest,

Why not?

Reply:
Honestly I'm unsure. The type does seem to recognize that it may not be set here, but there's no documentation stating why that is the case, and the `testResult.perfStats` seem to be what other reporters use (see: https://github.com/jest-community/jest-junit/blob/12da1a20217a9b6f30858013175319c1256f5b15/utils/buildJsonResults.js#L112)

My best guess is that it's what some old version of Jest did but no longer does... Either way, I've updated the check to fall back to `perfStats` and left the original `test.duration` check in there, since it doesn't seem to be hurting or helping. I'm okay with removing the original check if you want.

Reply:
I had more questions about this. We chatted about it offline and determined that Jest is perhaps neglecting to populate this field because it is not used in Jest's standard reporter.  Instead Jest displays timings at the very end of the run instead of interactively, and has not calculated the numbers yet at this point in time.  This resolved my concern that Heft might be somehow incorrectly implementing the Jest contract (i.e. _why should Heft need to measure test times, when that is supposed to be a feature of the test runner?_).